### PR TITLE
Update/swift 5

### PR DESCRIPTION
**Pull request by DanielStormApps**
## Summary
- Update for Swift 5 and Xcode 10.2 support

## Notes
- The new `Hashable` updates are outside my comfort zone. There may be a better way to satisfy the new requirements.
    - Seems to me the new `hash(into:)` is a replacement for `hashValue`: https://developer.apple.com/documentation/swift/hashable/2995575-hash

### CODE REVIEW

File: Sources/Internal.swift [hunk 1]
```diff
@@ -558,13 +558,14 @@ extension String {
...
     var sha1: String? {
-        guard let input = self.data(using: .utf8) else {
-            return nil
-        }
-        var hash = [UInt8](repeating: 0, count: Int(CC_SHA1_DIGEST_LENGTH))
-        input.withUnsafeBytes {
-            _ = CC_SHA1($0, CC_LONG(input.count), &hash)
+        guard let input = self.data(using: .utf8) else { return nil }
+        
+        let hash = input.withUnsafeBytes { (bytes: UnsafeRawBufferPointer) -> [UInt8] in
```
Comment (DanielStormApps) [inline Sources/Internal.swift]: @kean would you prefer an `#if compiler(>=5.0)` here?

Comment (kean) [reply Sources/Internal.swift]: Please, that would be ideal. I intend to keep Swift 4.2 support for the next year or so.

File: Sources/ImageRequest.swift [hunk 2]
```diff
@@ -264,34 +264,34 @@ public extension ImageRequest {
...
 internal extension ImageRequest {
     struct CacheKey: Hashable {
         let request: ImageRequest
-
-        var hashValue: Int {
+        
+        func hash(into hasher: inout Hasher) {
             if let customKey = request._ref.cacheKey {
-                return customKey.hashValue
+                hasher.combine(customKey)
```
Comment (kean) [inline Sources/ImageRequest.swift]: It seems that this changes the behavior of `CacheKey`, please add return or `else`.

**Comment by kean** [comment]
Thanks you, @DanielStormApps! I reviewed your changes and made a couple of comments.

Also `.travis.yml` needs an update.

File: Sources/Internal.swift (hunk 1 above)
Comment (DanielStormApps) [reply Sources/Internal.swift]: Updated. Added `#if swift(>=5.0)` check.

File: Sources/ImageRequest.swift (hunk 2 above)
Comment (DanielStormApps) [reply Sources/ImageRequest.swift]: Whoops! Good catch. Also updated `LoadKey`, had the same issue.

**Comment by DanielStormApps** [comment]
> Thanks you, @DanielStormApps! I reviewed your changes and made a couple of comments.
> 
> Also `.travis.yml` needs an update.

Updated. Not sure if I applied the changes you had intended. Seems the builds are failing except for 10.2 environments.

**Comment by kean** [comment]
@DanielStormApps merged this plus merged the changes to remove per-platform targets. So now instead of `Nuke iOS`, `Nuke macOS`, etc targets we have a single target. No duplication.

**Pull request merged**